detectable/src/main/java/com/synopsys/integration/detectable/detectables/git/parsing/model/GitConfig.java

```diff
@@ -0,0 +1,48 @@
+package com.synopsys.integration.detectable.detectables.git.parsing.model;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+// A List<GitConfigNode> is a rough parsing result. This class provides a better API.
+public class GitConfig {
+    // Only including remote and branch nodes since the core node is not being used.
+    private final List<GitConfigRemote> gitConfigRemotes;
+    private final List<GitConfigBranch> gitConfigBranches;
+
+    public static GitConfig fromGitConfigNodes(final List<GitConfigNode> gitConfigNodes) {
+        final List<GitConfigRemote> gitConfigRemotes = gitConfigNodes.stream()
+                                                           .filter(node -> node.getType().equals("remote"))
+                                                           .map(node -> {
+                                                               String remoteNodeName = node.getName().orElseThrow(() -> new IllegalArgumentException("Expected remote node to have a name."));
+                                                               String remoteNodeUrl = node.getProperty("url").orElseThrow(() -> new IllegalArgumentException("Expected remote node to have a url field."));
+                                                               String remoteNodeFetch = node.getProperty("fetch").orElseThrow(() -> new IllegalArgumentException("Expected remote node to have a fetch field."));
+                                                               return new GitConfigRemote(remoteNodeName, remoteNodeUrl, remoteNodeFetch);
+                                                           })
+                                                           .collect(Collectors.toList());
+
+        final List<GitConfigBranch> gitConfigBranches = gitConfigNodes.stream()
+                                                            .filter(node -> node.getType().equals("branch"))
+                                                            .map(node -> {
+                                                                String remoteNodeName = node.getName().orElseThrow(() -> new IllegalArgumentException("Expected branch node to have a name."));
+                                                                String remoteNodeRemote = node.getProperty("remote").orElseThrow(() -> new IllegalArgumentException("Expected branch node to have a remote field."));
+                                                                String remoteNodeMerge = node.getProperty("merge").orElseThrow(() -> new IllegalArgumentException("Expected branch node to have a fetch field."));
+                                                                return new GitConfigBranch(remoteNodeName, remoteNodeRemote, remoteNodeMerge);
+                                                            })
+                                                            .collect(Collectors.toList());
+
+        return new GitConfig(gitConfigRemotes, gitConfigBranches);
+    }
```

Comment:
Should this go in it's own class? It's only 22 lines and it only relevant to GitConfig which is why I added it here.

Reply:
I think I'd lean toward leaving it in that class as you have it since there's no other complexity in that class, but I could see arguments either way. Could the constructor become private?

Reply:
Having the constructor public allows for easier testing. Which isn't a great reason to affect source, but I think the impact is small here.

Reply:
ah, right.

Reply:
I'd very much prefer a GitConfigParser and a GitConfig data object.

Reply:
TheGitFileParser already exists. It converts the file into List<GitConfigNode>. The GitConfig model is meant to wrap the GitConfig nodes with a better API.

Reply:
I'd still prefer the operation of transforming some data be separate from the result of the transformation.